test/lit/passes/optimize-instructions-gc.wast

```diff
@@ -2728,34 +2707,65 @@
   ;; CHECK-NEXT: )
   ;; NOMNL:      (func $ref-cast-static-squared-impossible (type $eqref_=>_none) (param $x eqref)
   ;; NOMNL-NEXT:  (drop
-  ;; NOMNL-NEXT:   (ref.cast null $struct
-  ;; NOMNL-NEXT:    (ref.cast null $array
-  ;; NOMNL-NEXT:     (local.get $x)
+  ;; NOMNL-NEXT:   (ref.cast null none
+  ;; NOMNL-NEXT:    (local.get $x)
+  ;; NOMNL-NEXT:   )
+  ;; NOMNL-NEXT:  )
+  ;; NOMNL-NEXT:  (drop
+  ;; NOMNL-NEXT:   (block (result (ref $struct))
+  ;; NOMNL-NEXT:    (drop
+  ;; NOMNL-NEXT:     (ref.cast $array
+  ;; NOMNL-NEXT:      (local.get $x)
+  ;; NOMNL-NEXT:     )
   ;; NOMNL-NEXT:    )
+  ;; NOMNL-NEXT:    (unreachable)
```

Comment:
Would `vacuum` or some other pass get rid of the cast here since it's only effect is to trap, but it's followed by an unreachable anyway? We could do this even without --traps-never-happen.

Reply:
Hmm, atm no, we don't look forward in the block for this sort of thing.

In TNH mode the cast will just be removed anyhow due to the drop, so you're right this can be optimized, but only in non-TNH. For that reason it seems low priority to do, but in principle we could.